src/core/array.d

```diff
@@ -0,0 +1,61 @@
+module core.array;
+
+/++
+    Returns a static array constructed from `a`. The type of elements can be
+    specified implicitly (`int[2] a = staticArray(1,2)`) or explicitly
+    (`float[2] a = staticArray!float(1,2)`).
++/
+pragma(inline, true) U[T.length] staticArray(U = CommonType!T, T...)(T a)
+{
+    return [a];
+}
+
+unittest
+{
+    {
+        auto a = staticArray(1, 2, 3);
+        assert(is(typeof(a) == int[3]));
+        assert(a == [1, 2, 3]);
+    }
+    {
+        auto a = staticArray(1, 2.0);
+        assert(is(typeof(a) == double[2]));
+        assert(a == [1, 2.0]);
+    }
+    assert(!__traits(compiles, staticArray(1, "")));
+    assert(is(typeof(staticArray()) == void[0]));
+    // NOTE: `int[] temp=staticArray(1,2)` correctly issues a deprecation
+
+    {
+        auto a = staticArray!float(1, 2);
+        assert(is(typeof(a) == float[2]));
+        assert(a == [1, 2]);
+    }
+}
+
+package:
+// copied from std.traits ; TODO: move to core.internal.adapted?
```

Comment:
Well, imho `std.traits` and `std.meta` should have been part of druntime, but yeah for nor `core.internal.traits` is the place to dump traits from Phobos.